### PR TITLE
yubioath-flutter: init at 6.0.2

### PR DESCRIPTION
**Pull request by lukegb**
###### Description of changes

Add yubioath-flutter, the replacement for yubioath-desktop.

* Add Python bindings for `zxing-cpp`
* Add support in `mkFlutterApp` for applications which don't produce a `.packages` file for whatever reason
* Init https://github.com/Yubico/yubioath-flutter

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/zxing_cpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    numpy
+  ];
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  checkInputs = [
+    pillow
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/zxing_cpp/default.nix:30]: Where is meta?

File: pkgs/applications/misc/yubioath-flutter/helper.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  propagatedBuildInputs = [
+    poetry-core
+    yubikey-manager
+    fido2
+    mss
+    zxing_cpp
+    pillow
+    cryptography
+  ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yubioath-flutter/helper.nix:45]: meta is missing

File: pkgs/applications/misc/yubioath-flutter/helper.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    sed -i \
+      -e 's,zxing-cpp = .*,zxing-cpp = "*",g' \
+      -e 's,mss = .*,mss = "*",g' \
+      pyproject.toml
+  '';
+
+  postInstall = ''
+    mkdir -p $out/bin
+    cp authenticator-helper.py $out/bin/authenticator-helper
+    chmod +x $out/bin/authenticator-helper
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yubioath-flutter/helper.nix:31]: Why not use install?

File: pkgs/applications/misc/yubioath-flutter/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  buildInputs = [
+    pcre2
+  ];
+
+  meta = with lib; {
+    description = "Yubico Authenticator for Desktop";
+    homepage = "https://github.com/Yubico/yubioath-flutter";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ lukegb ];
+    platforms = builtins.attrNames vendorHashes;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yubioath-flutter/default.nix:56]:
```suggestion
    platforms = lib.attrNames vendorHashes;
```

File: pkgs/applications/misc/yubioath-flutter/default.nix [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+in
+flutter.mkFlutterApp rec {
+  pname = "yubioath-flutter";
+  inherit src version;
+
+  passthru.helper = helper;
+
+  vendorHash = vendorHashes."${stdenv.system}";
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yubioath-flutter/default.nix:32]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/misc/yubioath-flutter/default.nix [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  helper = python3.pkgs.callPackage ./helper.nix { inherit src version; };
+in
+flutter.mkFlutterApp rec {
+  pname = "yubioath-flutter";
+  inherit src version;
+
+  passthru.helper = helper;
+
+  vendorHash = vendorHashes."${stdenv.system}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yubioath-flutter/default.nix:30]: Missing or throw system

File: pkgs/applications/misc/yubioath-flutter/default.nix [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+let
+  vendorHashes = {
+    x86_64-linux = "sha256:0bwrybwaq9q16dwr56590k8j66qfckrjihsqj9brwjvlqc1mc207";
+  };
+
+  version = "6.0.2";
+  src = fetchFromGitHub {
+    owner = "Yubico";
+    repo = "yubioath-flutter";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/yubioath-flutter/default.nix:18]:
```suggestion
    rev = version;
```

**Comment by marsam** [comment]
@GrahamcOfBorg eval

File: pkgs/applications/misc/yubioath-flutter/helper.nix (hunk 2 above)
Comment (lukegb) [reply pkgs/applications/misc/yubioath-flutter/helper.nix:43]: Fixed.

File: pkgs/applications/misc/yubioath-flutter/helper.nix (hunk 3 above)
Comment (lukegb) [reply pkgs/applications/misc/yubioath-flutter/helper.nix]: No reason not to here. Let's do that.

File: pkgs/applications/misc/yubioath-flutter/default.nix (hunk 4 above)
Comment (lukegb) [reply pkgs/applications/misc/yubioath-flutter/default.nix]: This is not strictly necessary (or useful: the lib name is IMO only useful if it does something other than alias). nixpkgs (including lib/!) already includes many instances of just using builtins.attrNames; the lib alias is now really only useful for doing bulk `inherit (lib) ..... attrNames;`.

File: pkgs/applications/misc/yubioath-flutter/default.nix [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  '';
+
+  preInstall = ''
+    # Make sure we have permission to delete things CMake has copied in to our build directory from elsewhere.
+    chmod -R +w build/
+  '';
+  postInstall = ''
+    # Swap the authenticator-helper symlink with the correct symlink.
+    rm "$out/app/helper/authenticator-helper"
+    ln -s "${helper}/bin/authenticator-helper" "$out/app/helper/authenticator-helper"
```
[the remark this answers is not in the document]
Comment (lukegb) [reply pkgs/applications/misc/yubioath-flutter/default.nix]: I wanted to leave it like this so it actually explicitly fails if something has changed such that app/helper/authenticator-helper is missing, but to be honest I think that's not a particularly interesting failure mode anyway, so let's change it.

File: pkgs/applications/misc/yubioath-flutter/default.nix (hunk 7 above)
Comment (lukegb) [reply pkgs/applications/misc/yubioath-flutter/default.nix]: Good catch - I thought this was tagged as v${version} originally and forgot to unquote it once I noticed it wasn't :)

**Pull request merged**